cmake/LYPython.cmake

```diff
@@ -226,7 +226,7 @@ ly_set(LY_PACKAGE_VALIDATE_PACKAGE ${temp_LY_PACKAGE_VALIDATE_PACKAGE})
 
 if (NOT CMAKE_SCRIPT_MODE_FILE)
     # if we're in script mode, we dont want to actually try to find package or anything else
-
+    set (Python_ROOT_DIR ${LY_ROOT_FOLDER}/python/runtime/${LY_PYTHON_PACKAGE_NAME}/python)
```

Comment:
@kh-huawei I am assuming the version of python you have installed matches the current `LY_PYTHON_VERSION` value which is 3.10.5?

Reply:
My local python-versions  as of now are 3.11.1 and 3.10.7, and none of them are in my PATH (i.e. only installed in AppData/Local/...).

But upon some further testing I am not able to reproduce the issue that caused me to add this line.
I suspect this was a transient issue that occurred during the python upgrade, since back then cmake did find the wrong python at one point.

But since i can't reproduce the issue now on my end, should i close this MR?

Reply:
> > The way our scripts work is that they all explicitly run the python exe from the runtime folder, and python itself will use the python dlls and packages adjacent to its executable, so I'm not sure how it came to be that your build ran the wrong python unless someone had used some sort of operating system level override like PATH (or linux, LD_LIBRARY_PATH).
> 
> Hmm.
> 
> The LY_PYTHON_CMD always points to `${LY_ROOT_FOLDER}/python/runtime/python-3.10.5-rev1-windows/python/python.exe`, so i'm assuming that's the executable that was actually used.
> 
> But the `find_package(Python ...)` - call did return a different python version which would lead to the case where the wrong value for `${Python_BINFOLDER}` is appended to the `CMAKE_PROGRAM_PATH`.
> 
> With this change i try to avoid that by setting the `Python_ROOT_DIR` hint before the `find_package()` - call.
> 
> But since i can't really reproduce the issue anymore: Is it possible this behaviour was just an edge case, where the old path was still in the cmake cache or something?



> > The way our scripts work is that they all explicitly run the python exe from the runtime folder, and python itself will use the python dlls and packages adjacent to its executable, so I'm not sure how it came to be that your build ran the wrong python unless someone had used some sort of operating system level override like PATH (or linux, LD_LIBRARY_PATH).
> 
> Hmm.
> 
> The LY_PYTHON_CMD always points to `${LY_ROOT_FOLDER}/python/runtime/python-3.10.5-rev1-windows/python/python.exe`, so i'm assuming that's the executable that was actually used.
> 
> But the `find_package(Python ...)` - call did return a different python version which would lead to the case where the wrong value for `${Python_BINFOLDER}` is appended to the `CMAKE_PROGRAM_PATH`.
> 
> With this change i try to avoid that by setting the `Python_ROOT_DIR` hint before the `find_package()` - call.
> 
> But since i can't really reproduce the issue anymore: Is it possible this behaviour was just an edge case, where the old path was still in the cmake cache or something?



> > The way our scripts work is that they all explicitly run the python exe from the runtime folder, and python itself will use the python dlls and packages adjacent to its executable, so I'm not sure how it came to be that your build ran the wrong python unless someone had used some sort of operating system level override like PATH (or linux, LD_LIBRARY_PATH).
> 
> Hmm.
> 
> The LY_PYTHON_CMD always points to `${LY_ROOT_FOLDER}/python/runtime/python-3.10.5-rev1-windows/python/python.exe`, so i'm assuming that's the executable that was actually used.
> 
> But the `find_package(Python ...)` - call did return a different python version which would lead to the case where the wrong value for `${Python_BINFOLDER}` is appended to the `CMAKE_PROGRAM_PATH`.
> 
> With this change i try to avoid that by setting the `Python_ROOT_DIR` hint before the `find_package()` - call.
> 
> But since i can't really reproduce the issue anymore: Is it possible this behaviour was just an edge case, where the old path was still in the cmake cache or something?

Could it have been possible that at the time the issue occurred the O3DE Python version matched the System Python version?

Reply:
Please do close it if the change is no longer required, we get an email listing all the Pull Requests that are currently open.